providers/implementations/signature/eddsa_sig.c

```diff
@@ -161,6 +162,10 @@ int ed25519_digest_sign(void *vpeddsactx, unsigned char *sigret,
         ERR_raise(ERR_LIB_PROV, PROV_R_OUTPUT_BUFFER_TOO_SMALL);
         return 0;
     }
+    if (edkey->privkey == NULL) {
+        ERR_raise(ERR_LIB_PROV, EC_R_MISSING_PRIVATE_KEY);
```

Comment:
This is wrong, you cannot use EC_R with ERR_LIB_PROV. Use `PROV_R_NOT_A_PRIVATE_KEY` instead.